Duration arithmetic

ptomato opened the pull request:
Closes: #408

Ms2ger commented on docs/balancing.md:
```diff
@@ -0,0 +1,121 @@
+duration3 = duration1.minus(duration2, { disambiguation: 'balance' });
+
+tmp = duration1.minus(duration2);
+duration3 = tmp.with(tmp, { disambiguation: 'balance' });
```
One more for `from`

Ms2ger commented:
Travis issue is out of memory dealing with the test262 results. Looking for a solution.

Ms2ger commented:
Rebasing over https://github.com/tc39/proposal-temporal/pull/491 should help.

ptomato commented:
I have still mysteriously lost my merging privileges, but I do think I've addressed your comments from last round. Thanks for investigating the OOM error, I had overlooked it!
